### PR TITLE
Add now, daysInMonth. Enhance isLeapYear

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Implement the `now` and `daysInMonth` methods.
- Enhance `isLeapYear` and `formatsSeconds` methods.

### CODE REVIEW

File: src/Common/TimeZone.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ class TimeZone extends \DateTimeZone
     /**
      * Set Default timeZone.
      *
-     * @param (string) $tz valid time zone
```
Comment (peter279k) [inline src/Common/TimeZone.php:24]: Follow the PHPDoc comment annotation style and remove the `()`.

File: src/Common/TimeZone.php [hunk 2]
```diff
@@ -55,6 +55,6 @@ public static function tZIdentifiers() :array
      */
     public static function validate($tz) :string
     {
-        return (in_array($tz, static::tZIdentifiers()) === true) ? $tz : 'UTC';
+        return in_array($tz, static::tZIdentifiers()) ? $tz : 'UTC';
```
Comment (peter279k) [inline src/Common/TimeZone.php:58]: The `in_array` will return bool value, and removing the `=== true` statement makes sense.

File: src/Time/Time.php [hunk 3]
```diff
@@ -16,118 +16,111 @@
...
      *
      * @param string $timezone Valid php supported timezone.
      *
      * @since 3.0.0
      *
-     * @return  int
+     * @return int
      */
-    public static function now($timezone = NULL)
+    public static function now($timezone = '')
```
Comment (peter279k) [inline src/Time/Time.php:33]: The `$timezone` will be the `string` type, not `null`.

And it makes sense that let the `$timezone` be empty string by default.

File: src/Time/Time.php [hunk 4]
```diff
@@ -16,118 +16,111 @@
...
      *
      * @since 3.0.0
      *
-     * @return  int
+     * @return int
      */
     public static function daysInMonth($month = 1, $year = 1970)
     {
-
+        return cal_days_in_month(CAL_GREGORIAN, $month, $year);
```
Comment (peter279k) [inline src/Time/Time.php:52]: Using the `cal_days_in_month` function which is provided by `calendar` extension in PHP to calculate the days of given month.

File: src/Time/Time.php [hunk 5]
```diff
@@ -16,118 +16,111 @@
...
+     */
     public static function isLeapYear($year = null)
     {
-        if (null === $year) {
-            $year = date('y');
-        }
-       if ((($year % 4 == 0) && ($year % 100 != 0)) || ($year % 400 == 0)) {
-            return true;
-       }
+        $year = $year ?? date('Y');
```
Comment (peter279k) [inline src/Time/Time.php:66]: Using the `??` syntax to check whether the `$year` is assigned.

File: src/Time/Time.php [hunk 6]
```diff
@@ -16,118 +16,111 @@
...
 
-       return false;
+        return (($year % 4 == 0) && ($year % 100 != 0)) || ($year % 400 == 0);
```
Comment (peter279k) [inline src/Time/Time.php:68]: Remove the `if` condition. Using the `?:` syntax  instead.

File: src/Time/Time.php [hunk 7]
```diff
@@ -16,118 +16,111 @@
...
-     * @param (int) $time Timestamp
+     * @param int    $time     Timestamp
+     * @param string $timezone Valid php supported timezone.
      *
      * @since 3.0.0
      *
      * @return mixed
      */
-    public static function formatsSeconds($seconds)
+    public static function formatsSeconds($seconds, $timezone = '')
```
Comment (peter279k) [inline src/Time/Time.php:120]: We should give the `$timezone` name to set timezone before formatting the seconds.

And the default timezone is `UTC` if the `$timezone` is empty.

File: src/Time/Time.php [hunk 8]
```diff
@@ -16,118 +16,111 @@
...
     {
-        $h = ($seconds >= 3600) ? (int) round($seconds / 3600) : 0;
-        $time = ($seconds >= 3600) ? $seconds % 3600 : $seconds;
-        $m = (int) $time / 60;
-        $s = (int) $time % 60;
+        TimeZone::seteDefaultTz($timezone);
 
-        return (int) $h.':'.(int) $m.':'.(int) $s;
+        return date('h:i:s', $seconds);
```
Comment (peter279k) [inline src/Time/Time.php:124]: Using the native `date` function in PHP to format seconds.

I think this approach will be more efficient than using own second calculation.

**Comment by lablnet** [comment]
@peter279k  thanks you so much

**Pull request merged**